it/AssetsControllerISpec.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.net.URLEncoder
+
+import helpers.IntegrationSpecBase
+import javax.wsdl.xml.WSDLReader
+import javax.wsdl.{Operation, PortType}
+import org.apache.axis2.wsdl.WSDLUtil
+
+import scala.collection.JavaConverters._
+
+class AssetsControllerISpec extends IntegrationSpecBase {
+
+  val wsdlBaseUrl = s"http://localhost:$port/assets/wsdl/"
+
+  val wsdlOperationsForFileNames = Map(
+    "CCN2.Service.Customs.Default.ICS.ReferralManagementBAS_1.0.0_CCN2_1.0.0.wsdl" -> List(
+      "IE4Q04requestAdditionalInformation",
+      "IE4R02provideAdditionalInformation",
+      "IE4Q05requestHRCM",
+      "IE4R03provideHRCMResult"
+    ),
+    "CCN2.Service.Customs.Default.ICS.RiskAnalysisOrchestrationBAS_1.0.0_CCN2_1.0.0.wsdl" -> List(
+      "IE4N03notifyERiskAnalysisHit",
+      "IE4S01submitEScreeningAssessment",
+      "IE4S02submitRiskAnalysisResult",
+      "IE4S02updateERiskAnalysisResult",
+      "IE4S01updateEScreeningResult"
+    )
+  )
+
+  wsdlOperationsForFileNames.map {
+    case (fileName, wsdlOperationList) =>
+      s"/assets/wsdl/$fileName" when {
+        val wsdlUrl = wsdlBaseUrl + URLEncoder.encode(fileName, "UTF-8")
+        s"a request is made for $fileName" should {
+          val operations = parseWsdlAndGetOperationsNames(wsdlUrl)
```

Comment:
Do we need to file encode?
```suggestion
      s"a request is made to /assets/wsdl/$fileName" when {
        val operations = parseWsdlAndGetOperationsNames(wsdlBaseUrl + fileName)
```

Reply:
im wondering about this, surely it could cause issues 🤔 , 
i have no idea what the wsdl library does,
sure a file name could have spaces in, and other weird characters, and we've just been lucky
maybe we will have to url encode and decode if we get issues when the stub does all of its cascading calls

Reply:
I think if we get the requirement we'll need to do it -> if it's not the initial wsdl then the library handles the retrieval of the files. If we have a problematic file either we could encode in outbound proxy and decode here. Until we have the requirement or need I don't think it's a good idea to only decode :+1:

Reply:
> There is nothing in the story to relate to the structuring of the files. I looked at the structure and they are very long with ~ symbols. The filenames are different so it made more sense to have them at top level rather than have huge complex urls to pass around everywhere.

its been missed off when the story was split out, the story was to put the files in the structure of the eu zip file, in the correct path structure (relative path structure)

Reply:
see history of ticket in jira